clang/lib/StaticAnalyzer/Core/SimpleSValBuilder.cpp

```diff
@@ -538,9 +553,12 @@ SVal SimpleSValBuilder::evalBinOpNN(ProgramStateRef state,
           CompareType.apply(LHSValue);
           CompareType.apply(RHSValue);
         } else if (!BinaryOperator::isShiftOp(op)) {
-          APSIntType IntType = BasicVals.getAPSIntType(resultTy);
-          IntType.apply(LHSValue);
-          IntType.apply(RHSValue);
+          // Check if resultTy is valid before using it
+          if (!resultTy.isNull()) {
```

Comment:
Checking the `resultTy` here and there without systematic guarantees doesn't seem like the right approach.
We should think about making this bulletproof somehow.